packages/ckeditor5-utils/src/dom/position.ts

```diff
@@ -99,12 +100,44 @@ export function getOptimalPosition( { element, target, positions, limiter, fitIn
 
 	let bestPosition: Position;
 
+	// Get intersection of all scrollable ancestors of `target`.
+	const allScrollableAncestors = getScrollableAncestors( target as HTMLElement );
+	const ancestorsIntersection: Rect | null = allScrollableAncestors
+		.reverse()
```

Comment:
Out of curiosity, why reverse and go from the root rect to the leaf rect instead the opposite (default) direction here?

Reply:
Need to dig deeper in my memory (shame on me that i didn't comment it). TBH was thinking about it when i create this PR.